src/precompile_MixedModels.jl

```diff
@@ -0,0 +1,77 @@
+#! format: off
+# this file is automatically generated by SnoopCompile
+
+function _precompile_()
+    ccall(:jl_generating_output, Cint, ()) == 1 || return nothing
+    let fbody = try Base.bodyfunction(which(fit!, (LinearMixedModel,))) catch missing end
+    if !ismissing(fbody)
+        precompile(fbody, (Bool,Bool,Nothing,Int64,typeof(fit!),LinearMixedModel,))
+    end
+end   # time: 6.582904
+    Base.precompile(Tuple{typeof(fit!),GeneralizedLinearMixedModel})   # time: 6.0995073
+    Base.precompile(Tuple{Type{LinearMixedModel},AbstractArray,Tuple,FormulaTerm,Vector{Any},Nothing})   # time: 2.2513552
+    let fbody = try Base.bodyfunction(which(fit!, (LinearMixedModel{Float64},))) catch missing end
+    if !ismissing(fbody)
+        precompile(fbody, (Bool,Bool,Nothing,Int64,typeof(fit!),LinearMixedModel{Float64},))
+    end
+end   # time: 2.1690955
+    isdefined(MixedModels, Symbol("#22#23")) && Base.precompile(Tuple{getfield(MixedModels, Symbol("#22#23"))})   # time: 0.70335937
+    Base.precompile(Tuple{Type{LinearMixedModel},Vector{Float64},Tuple{Matrix{Float64}, ReMat{Float64, 2}},FormulaTerm{ContinuousTerm{Float64}, Tuple{MatrixTerm{Tuple{InterceptTerm{true}, ContinuousTerm{Float64}}}, RandomEffectsTerm}},Vector{Any},Nothing})   # time: 0.6737979
+    Base.precompile(Tuple{typeof(deviance!),GeneralizedLinearMixedModel,Int64})   # time: 0.6433022
+    Base.precompile(Tuple{Type{LinearMixedModel},Vector{Float64},FeTerm{Float64, Matrix{Float64}},Vector{AbstractReMat{Float64}},FormulaTerm{ContinuousTerm{Float64}, Tuple{MatrixTerm{Tuple{InterceptTerm{true}, ContinuousTerm{Float64}}}, RandomEffectsTerm}},Vector{Any},Nothing})   # time: 0.42538697
+    Base.precompile(Tuple{typeof(deviance!),GeneralizedLinearMixedModel{Float64, Bernoulli{Float64}},Int64})   # time: 0.3716886
+    Base.precompile(Tuple{typeof(_ranef_refs),InteractionTerm{Tuple{CategoricalTerm{StatsModels.FullDummyCoding, String, 2}, CategoricalTerm{StatsModels.FullDummyCoding, String, 60}}},NamedTuple{(:use, :age, :urban, :livch, :dist), Tuple{Vector{String}, Vector{Float64}, Vector{String}, Vector{String}, Vector{String}}}})   # time: 0.15686166
+    Base.precompile(Tuple{typeof(fit!),GeneralizedLinearMixedModel{Float64, Bernoulli{Float64}}})   # time: 0.1559177
+    Base.precompile(Tuple{typeof(modelcols),RandomEffectsTerm,NamedTuple{(:reaction, :days, :subj), Tuple{Vector{Float64}, Vector{Int8}, Vector{String}}}})   # time: 0.14000398
```

Comment:
Looks like this line and 44 are overly specific to the calls used to generate this, as the particular `NamedTuple` table structure is part of the method signature that's being precompiled. When testing the timing improvements I would try with a different dataset to see how much of an impact there is.

I've never used SnoopCompile before; are the timings provided in the comments at the end of each line the amount of time it took to execute the `precompile` call? If so, we're adding 0.157 seconds to precompilation time that will only benefit two specific examples. Obviously not a ton of time but figured it's worth noting.

Reply:
Good catch! And super interesting to see how the Tables.jl representation plays through these things. I had also noticed that SnoopCompile had a lot to say about StatsModels that was quite specific, but then again StatsModels is known for hitting the compiler hard.